### ISSUE TITLE
Hardcoded "--cpu 5" in run_dbcan.py

### ISSUE THREAD

**Issue by cmkobel** [linnabrown/run_dbcan#121]
Hello 
I was messing around trying to optimize how dbcan runs on my machine and I was testing different counts of CPU for multithreading.

I set all my CPU counts to 3, expecting to see that all the subsequent processes would use 3 (possibly 4 when a master thread is running, as is the case for "hmmscan")

This is my call
```
run_dbcan \
    --dbcan_thread {threads} \
    --dia_cpu {threads} \
    --hmm_cpu {threads} \
    --tf_cpu {threads} \
    --stp_cpu {threads} \
    --db_dir $(dirname {input.database_representative}) \
    --out_dir {params.out_dir} \
    {input.aminoacid} \
    protein
```

But, when I run dbcan (dbcan 4.0.0 pyh7cba7a3_0 bioconda) I see that hmmscan is called with 5 cpus:

```
pgrep -a hmmscan
#> 3520575 hmmscan --domtblout results_ac2/samples/Control_HighE.vamb.4912/dbcan/d0.txt --cpu 5 -o /dev/null /home/thylakoid/assemblycomparator2/databases/dbcan/dbCAN_sub.hmm results_ac2/sample /Control_HighE.vamb.4912/dbcan/0.txt
```

So I started lurking around in the code base and I see that on the following lines in run_dbcan.py, the set --hmm_cpu argument is not used. Instead a hard coded '5' is put into the hmmscan --cpu argument.
  - https://github.com/linnabrown/run_dbcan/blob/f3dd111a9eaf552cdca14f4c8db06baabaa1f2f8/dbcan_cli/run_dbcan.py#L90
  - https://github.com/linnabrown/run_dbcan/blob/f3dd111a9eaf552cdca14f4c8db06baabaa1f2f8/dbcan_cli/run_dbcan.py#L114

I think it would be absolutely splendid if one of the developers would change this to use the --hmm_cpu argument instead, so we can unleash the full potential of extremely highly cored CPUs and find all the CAZymes and substrates that every single sequenced microbe utilizes.

Best, Carl

**Comment by chtsai0105** [comment]
Hi - I've also found a related issue in #117. The hardcoded "--cpu 5" also causes cpu over-spawning issue when running several jobs parallelly.

The thing is, the hmmscan not that cpu efficient. According to this [article](http://cryptogenomicon.org/hmmscan-vs-hmmsearch-speed-the-numerology.html), what major limitation of hmmscan is input speed but not cpu speed. We also observed that the single cpu usage is extremely low while multithreading. So I guess that the main reason they constrain the cpu to 5.
